packages/field/src/LionField.js

```diff
@@ -178,14 +177,12 @@ export class LionField extends FormControlMixin(
     );
   }
 
-  _onValueChanged({ value }) {
-    if (super._onValueChanged) {
-      super._onValueChanged();
+  _onModelValueChanged({ modelValue }) {
+    if (super._onModelValueChanged) {
+      super._onModelValueChanged();
     }
     // For styling purposes, make it known the input field is not empty
-    if (this._inputNode) {
-      this[value ? 'setAttribute' : 'removeAttribute']('filled', '');
-    }
+    this[modelValue !== '' ? 'setAttribute' : 'removeAttribute']('filled', '');
```

Comment:
Totally agree we should base derived state always on. modelValue, since this is our internal SSOT.

This might actually be cleaner (and better for js engine optimalization(v8 etc)) to do it in a regular if/else.

Or we could maybe just make it a property with `reflect:true`?

Reply:
We could also consider putting this logic in interactionState.

Also, ideally an empty modelValue should be `undefined` instead of ''. (I think there was a mr for this a while ago, but maybe we never followed through)

Reply:
Empty string would still parse to a modelValue -> '' right? It doesn't seem right to me that a parser ever returns undefined, unless the literal input you give it is JavaScript's primitive: `undefined`. `null` would be more correct than `undefined` in my opinion, but I consider `null` to also not be as appropriate as empty string.

And yes I think it makes sense to make it a property with reflect: true and putting it inside InteractionStateMixin :) I can do that

Reply:
Ah, didn't see your response. I changed my initial reaction a bit in the meantime... :)
'' assumes modelValue is a String, which we can't do.

I was also in favor of `null` before, but it was a team decision to go for `undefined` in the end. The nice thing is that it is similar to an initally empty value.

Reply:
Alright makes sense, and I agree with "The nicest solution", and then filled is only an attribute, just as a public styling hook. `_isEmpty` should be used for logic by subclassers, and it is used to set the filled styling hook by doing in InteractionStateMixin on model value changed:
```js
if (this._isEmpty) {
  this.removeAttribute('filled');
} else {
  this.setAttribute('filled');
}
``` 
Seems ok?

The reason I don't think we need filled to be a lit property is because 1) it's not a complex value so attr is fine and 2) we won't need to re-render the component since it's just meant as a styling hook (using css attr selector) which doesn't require a re-render.

Reply:
Cool that we agree 😎
About the lit prop: It won't trigger a rerender of the lit template, but the attribute is used as a styling hook for the outside world. Whenever there is styling for `[filled]`, a new paint will be triggered by the browser. It's a better practice to batch all those attrs in a batch I think (like LitElement does for you).
On top of that, I think it's nice to have an overview of all InteractionState properties that, when our api tables are in place, will automatically be documented.

Reply:
Ah I will make it a lit property. I never considered that making a style hook attr a lit property gives the benefit of ending up in a batched paint, that's a really cool feature (maybe side effect) of lit actually :O